content/api/configuration.md

```diff
@@ -1,10 +1,1407 @@
+
+  [stats](#stats): {
+    /* TODO */
+  },
+
+  [devServer](#devServer): {
+    /* TODO */
+  },
+
+  [plugins](#plugins): [
+    // ...
+  ],
+  // list of additional plugins
+
+  <details><summary>/* Advanced configuration (click to show) */</summary>
+
+  [resolveLoader](#resolveLoader): { /* same as resolve */ }
+  // separate resolve options for loaders
+
+  [profile](#profile): true, // boolean
+  // capture timing information
+
+  [cache](#cache): false, // boolean
+  // disable/enable caching
+
+  [watch](#watch): true, // boolean
+  // enables watching
+
+  [watchOptions](#watchOptions): {
+    [aggregateTimeout](#watchOptions.aggregateTimeout): 1000, // in ms
+    // aggregates multiple changes to a single rebuild
+
+    [poll](#watchOptions.poll): true,
+    [poll](#watchOptions.poll): 500, // intervall in ms
+    // enables polling mode for watching
+    // must be used on filesystems that doesn't notify on change
+    // i. e. nfs shares
+  },
+
+  [node](#node): {
+    /* TODO */
+  },
+
+  [recordsPath](#recordsPath): path.resolve(__dirname, "build/records.json"),
+  [recordsInputPath](#recordsInputPath): path.resolve(__dirname, "build/records.json"),
+  [recordsOutputPath](#recordsOutputPath): path.resolve(__dirname, "build/records.json"),
+  // TODO
+
+  </details>
+}
+```
+
+---
+
+
+### `context`
+
+`string`
+
+The base directory, an absolute path, for resolving entry points.
+
+```js
+context: __dirname + ‘/src’
+```
+
+---
+
+### `entry`
+
+`string` `array` `object`
+
+The point or points to enter the application. 
+
+```js
+entry: {
+  home: ‘./home.js’,
+  about: ‘./about.js’,
+  contact: ‘./contact.js’
+}
+```
+
+?> TODO: Add more details on passing a string/array and how this will affect bundle names
+
+---
+
+### `output`
+
+`object`
+
+A set of options instructing webpack on how and where it should output your bundles, assets and anything else you bundle up or load with webpack. Below is a simple example followed by descriptions of each individual `output` option.
+
+```js
+output: {
+  filename: ‘[name].bundle.js’,
+  path: __dirname + ‘/dist/assets’,
+  publicPath: ‘/assets/’
+}
+```
+
+---
+
+### `output.filename`
+
+`string`
+
+This option determines the name of each output bundle. For a single `entry` point with no children, this can be a static name. However, when creating multiple bundles via more than one entry point, [code splitting](), or various [plugins](), you should use one of the following substitutions to give each bundle a unique name...
+
+Using entry/chunk names:
+```js
+filename: [name].bundle.js
+```
+
+Using the unique hash generated for every build:
+```js 
+filename: [name].[hash].bundle.js
+```
+
+Using hashes based on each chunks' content:
+```js
+filename: [chunkhash].bundle.js
+```
+
+---
+
+### `output.path`
+
+`string`
+
+The output directory as an **absolute** path. Note that the `[hash]` subsitution may be used here as well.
+
+```js
+path: __dirname + '/dist/assets'
```

Comment:
True. `path.resolve` is even better.
`path.resolve(__dirname, "dist/assets")` vs `path.join(__dirname, "dist", "assets")`

`__dirname + '/...` should never be used in documentation!